src/main/java/com/github/stefvanschie/inventoryframework/GuiItem.java

```diff
@@ -33,6 +35,12 @@
      */
     @Nullable
     private Consumer<InventoryClickEvent> action;
+    
+    /**
+     * List of item's properties
+     */
+    @Nullable
+    private List<Object> properties;
```

Comment:
Probably better to always have a value for `properties` and just have an empty list when there are no properties: this avoids having to do null checks; `@Nullable` can then be changed to `@NotNull`.

Reply:
My bad, should've used `@NotNull` right away, since `properties` is initialized with an empty list inside of the constructor already, will fix  with upcoming commit